model/src/pyrenew/latent/infection_seeding_method.py

```diff
@@ -0,0 +1,175 @@
+            raise TypeError(
+                f"n_timepoints must be an integer. Got {type(n_timepoints)}"
+            )
+        if n_timepoints <= 0:
+            raise ValueError(
+                f"n_timepoints must be positive. Got {n_timepoints}"
+            )
+
+    @abstractmethod
+    def seed_infections(self, I_pre_seed: ArrayLike):
+        """Generate the number of seeded infections at each time point.
+
+        Parameters
+        ----------
+        I_pre_seed : ArrayLike
+            An array representing some number of latent infections to be used with the specified ``InfectionSeedMethod``.
+
+        Returns
+        -------
+        ArrayLike
+            An array of length ``n_timepoints`` with the number of seeded infections at each time point.
+        """
+        pass
+
+    def __call__(self, I_pre_seed: ArrayLike):
+        return self.seed_infections(I_pre_seed)
+
+
+class SeedInfectionsZeroPad(InfectionSeedMethod):
+    """Pad the seed infections with zeros at the beginning of the time series."""
+
+    def seed_infections(self, I_pre_seed: ArrayLike):
+        """Pad the seed infections with zeros at the beginning of the time series.
+
+        Parameters
+        ----------
+        I_pre_seed : ArrayLike
+            An array with seeded infections to be padded with zeros.
+
+        Returns
+        -------
+        ArrayLike
+            An array of length ``n_timepoints`` with the number of seeded infections at each time point.
+        """
+        if self.n_timepoints < I_pre_seed.size:
+            raise ValueError(
+                f"I_pre_seed must be at least as long as n_timepoints. Got I_pre_seed of size {I_pre_seed.size} and and n_timepoints of size {self.n_timepoints}."
+            )
+        # alternative implementation:
+        # return jnp.hstack([jnp.zeros(self.n_timepoints - I_pre_seed.size), I_pre_seed])
+        return jnp.pad(I_pre_seed, (self.n_timepoints - I_pre_seed.size, 0))
+
+
+class SeedInfectionsFromVec(InfectionSeedMethod):
+    """Create seed infections from a vector of infections."""
+
+    def seed_infections(self, I_pre_seed: ArrayLike):
+        """Create seed infections from a vector of infections.
+
+        Parameters
+        ----------
+        I_pre_seed : ArrayLike
+            An array with the same length as ``n_timepoints`` to be used as the seed infections.
+
+        Returns
+        -------
+        ArrayLike
+            An array of length ``n_timepoints`` with the number of seeded infections at each time point.
+        """
+        if I_pre_seed.size != self.n_timepoints:
+            raise ValueError(
+                f"I_pre_seed must have the same size as n_timepoints. Got I_pre_seed of size {I_pre_seed.size} and and n_timepoints of size {self.n_timepoints}."
+            )
+        return jnp.array(I_pre_seed)
+
+
+class SeedInfectionsExponential(InfectionSeedMethod):
+    """Generate seed infections according to exponential growth.
```

Comment:
Needs to be a raw string or backslashes need to be escaped.
```suggestion
    r"""Generate seed infections according to exponential growth.
```